pandas/tests/io/json/test_ujson.py

```diff
@@ -40,116 +68,113 @@ def test_encodeDecimal(self):
         assert decoded == 1.0
 
-    def test_encodeStringConversion(self):
-        input = "A string \\ / \b \f \n \r \t </script> &"
+    @pytest.mark.parametrize("ensure_ascii", [True, False])
+    def test_encode_string_conversion(self, ensure_ascii):
+        string_input = "A string \\ / \b \f \n \r \t </script> &"
         not_html_encoded = ('"A string \\\\ \\/ \\b \\f \\n '
                             '\\r \\t <\\/script> &"')
         html_encoded = ('"A string \\\\ \\/ \\b \\f \\n \\r \\t '
                         '\\u003c\\/script\\u003e \\u0026"')
 
         def helper(expected_output, **encode_kwargs):
-            output = ujson.encode(input, **encode_kwargs)
-            assert input == json.loads(output)
+            output = ujson.encode(string_input,
+                                  ensure_ascii=ensure_ascii,
+                                  **encode_kwargs)
+
             assert output == expected_output
-            assert input == ujson.decode(output)
+            assert string_input == json.loads(output)
+            assert string_input == ujson.decode(output)
 
         # Default behavior assumes encode_html_chars=False.
-        helper(not_html_encoded, ensure_ascii=True)
-        helper(not_html_encoded, ensure_ascii=False)
+        helper(not_html_encoded)
 
         # Make sure explicit encode_html_chars=False works.
-        helper(not_html_encoded, ensure_ascii=True, encode_html_chars=False)
-        helper(not_html_encoded, ensure_ascii=False, encode_html_chars=False)
+        helper(not_html_encoded, encode_html_chars=False)
 
         # Make sure explicit encode_html_chars=True does the encoding.
-        helper(html_encoded, ensure_ascii=True, encode_html_chars=True)
-        helper(html_encoded, ensure_ascii=False, encode_html_chars=True)
+        helper(html_encoded, encode_html_chars=True)
 
-    def test_doubleLongIssue(self):
-        sut = {u('a'): -4342969734183514}
-        encoded = json.dumps(sut)
-        decoded = json.loads(encoded)
-        assert sut == decoded
+    @pytest.mark.parametrize("long_number", [
+        -4342969734183514, -12345678901234.56789012, -528656961.4399388
+    ])
+    def test_double_long_numbers(self, long_number):
+        sut = {u("a"): long_number}
         encoded = ujson.encode(sut, double_precision=15)
-        decoded = ujson.decode(encoded)
-        assert sut == decoded
 
-    def test_doubleLongDecimalIssue(self):
-        sut = {u('a'): -12345678901234.56789012}
-        encoded = json.dumps(sut)
-        decoded = json.loads(encoded)
-        assert sut == decoded
-        encoded = ujson.encode(sut, double_precision=15)
         decoded = ujson.decode(encoded)
         assert sut == decoded
 
-    def test_encodeNonCLocale(self):
+    def test_encode_non_c_locale(self):
         import locale
-        savedlocale = locale.getlocale(locale.LC_NUMERIC)
+        saved_locale = locale.getlocale(locale.LC_NUMERIC)
+
         try:
```

Comment:
i think we have a contest manager for locale setting

Reply:
* We do, but it doesn't do the `Exception` checking that I think this test wants to do.
* I think I would want to tap into `_can_set_locale` instead.  Any reason why I shouldn't utilize into this private method?

Reply:
not familiar with that
but happy to use existing code (so u know why it’s private?)